beacon-chain/blockchain/testing/mock.go

```diff
@@ -203,7 +205,7 @@ func (ms *ChainService) ReceiveBlockBatch(ctx context.Context, blks []*ethpb.Sig
 // ReceiveBlock mocks ReceiveBlock method in chain service.
 func (ms *ChainService) ReceiveBlock(ctx context.Context, block *ethpb.SignedBeaconBlock, _ [32]byte) error {
 	if ms.State == nil {
-		ms.State = &stateTrie.BeaconState{}
+		ms.State = testutil.NewBeaconState()
```

Comment:
I changed this because initializing the state with `&stateTrie.BeaconState{}`always fails this later check:
```
if err := ms.State.SetSlot(block.Block.Slot); err != nil {
    return err
}
```

Reply:
This change is pretty substantial no? Why did the previous code break?

Reply:
It's `beacon-chain/blockchain/testing/mock.go `, so the impact is test-only. The previous code did not work with my test because some field of the structure was `nil`, cause the check to fail.

Reply:
I see this broke one test, though... Removing the review label for now.